JacquardToolkit/JacquardToolkit/JacquardService.swift

```diff
@@ -209,33 +217,39 @@ extension JacquardService: CBPeripheralDelegate {
             input_data![JSConstants.JSNumbers.ForceTouch.fullThreadCount - JSConstants.JSNumbers.ForceTouch.threadCount + i] = threadReadings[i] as NSNumber
         }
         
-        let prediction = try? forceTouchModel.prediction(input: NewGestureClassifier_RC2Input(_15ThreadConductivityReadings: input_data!))
-        print("Prediction Result: \((prediction?.output["ForceTouch"])!)")
-        
+        let prediction = try? forceTouchModel.prediction(input: ForceTouchInput(_15ThreadConductivityReadings: input_data!))
+//        print("Prediction Result: \((prediction?.output["ForceTouch"])!)")
+
         if forceTouchTurnedEnabled {
-            if ((prediction?.output["ForceTouch"])! > 0.7) {
-                forceTouchTurnedEnabled = false
-                delegate?.didDetectForceTouchGesture()
-                return true
+            if ((prediction?.output[JSConstants.JSStrings.ForceTouch.outputLabel])! > forceTouchDetectionThreshold) {
+                // increment detection progress with each sufficiently high prediction confidence
+                forceTouchDetectionProgress += 1
+                // if enough detection progress has elapsed, register a detection of ForceTouch and reset
+                if forceTouchDetectionProgress >= forceTouchDetectionLength {
+                    forceTouchDetectionProgress = 0
+                    forceTouchTurnedEnabled = false
+                    delegate?.didDetectForceTouchGesture!()
+                    return true
```

Comment:
any reason why we still need this boolean @calebrudnicki , it's not used at the moment- i just kept it thus far since you added the method header as such:
private func checkForForceTouch(threadReadings: [Float]) -> **Bool** {

Reply:
Go ahead and take it out of the method signature and the class as a whole.